src/client.c

```diff
@@ -1,46 +1,206 @@
 #include "client.h"
 #include <assert.h>
+#include <pthread.h>
 #include <stdlib.h>
+#include <time.h>
+#include <stdio.h>
 
+// Diferentes niveles de bloqueo en reservas.
+static pthread_mutex_t global_mutex = PTHREAD_MUTEX_INITIALIZER;
+static pthread_mutex_t* row_mutexes;
+static pthread_mutex_t* seat_mutexes;
+
+// Inicializa Datos del cliente
 void client_data_init(client_data_t* data, movie_t* movie, int id) {
     data->movie = movie;
-    data->id    = id;
+    data->id = id;
+}
+
+// Crea un nuevo resultado de reserva.
+client_res_t* client_res_new() {
+    client_res_t* res = (client_res_t*)malloc(sizeof(client_res_t));
+    if (res == NULL) {
+        return NULL;
+    }
+    res->success = false;
+    return res;
 }
 
-void client_res_init(client_res_t* res) {
-    res->success = false; // Por defecto la reserva comienza fallando.
+// Libera un resultado de reserva.
+void client_res_free(client_res_t* res) {
+    free(res);
 }
 
-void* client_run(client_t* client) {
-    // De momento el cliente no hace nada.
-    return NULL;
+// Puntero a función. Se asignará a la función de ejecución que le corresponda.
+client_res_t* (*client_run)(client_data_t*) = NULL;
+
+// Bloqueo Global
+client_res_t* client_global_mutex(client_data_t* data) {
+    movie_t* movie = data->movie;
+    int id = data->id;
+
+    static int seed = 0xdeadbeef;
+    int row = rand_r(&seed) % movie->nrows;
+    int col = rand_r(&seed) % movie->ncols;
+
+    pthread_mutex_lock(&global_mutex);
+    client_res_t* res = client_res_new();
```

Comment:
Esto no hace falta que se reserve en la sección crítica, cada hebra reserva la suya propia y por lo tanto no puede tener condiciones de carrera.

Reply:
También hace falta chequear si `res` es NULL y fallar en esta condición.
```suggestion
    client_res_t* res = client_res_new();
    if (res == NULL) {
        return NULL;
    }
```

Reply:
Estos últimos dos comentarios aplican en las tres implementaciones de client_run